### DISCUSSION TITLE
Improving the RDKit's support for organometallics Part 1: Use cases

### DISCUSSION THREAD

**Discussion by greglandrum** [rdkit/rdkit#3618] [Development]
# A respectful request
If you are planning on taking part in this discussion, please read this first message in its entirety before participating. I really want to gather community feedback on this, but it's very, very easy for this kind of thing to run off the rails, and none of us have the time for that.

# Background
During this year's RDKit UGM we had a pretty lively discussion about organometallics and the RDKit. This was catalyzed (hahah) by Jan Jensen's "Organometallica" presentation ([slides here](https://github.com/rdkit/UGM_2020/raw/master/Presentations/JanJensen.pdf), [recording here](https://www.youtube.com/watch?v=HD6IpXMVKeo)). The discussion in discord kind of [starts here](https://discord.com/channels/753872808882470974/763004759627268116/763226771931660309).

I would like to make some real progress on this in early 2021 and in order to do so want to start collecting input and thoughts from the community. I originally had thought about doing this via email, and we'll fall back on that if we really need to, but I'd really like to have the discussion more publicly available and closer to the actual RDKit code which will result from it. Thus this discussion thread...

Here's what I wrote in that discord thread:

> I think it's worth starting a longer conversation about representing organometallics in the RDKit and what the use cases there are. This has been one of those things that's difficult to organize because everyone comes at it from a slightly different perspective and the "conversations" end up with people talking past each other instead of converging on a solution; that's a frustrating waste of everyone's time. We would start by avoiding all discussion about file formats and focus on the use cases for what people want to actually do with OM species in the RDKit (for example: "I want a canonical representation to store in my database" is very, very different from "I want to be able to do standard electron counting at the metal center to determine oxidation state and whether or not this is likely to be a stable species" and that's different from "I want to generate fingerprints so that I can do machine learning", which is different from "I want to generate reasonable 3D conformers"). From there we could talk about a bit about the science and what we think the molecular graph should look like, and from there we could start to consider file formats. Given that we're not all going to be sitting in one room for the foreseeable future, I'm not sure what the best way to do that conversation is, but I'd like to try something. 

# This discussion
The goal of this part of the discussion is to answer questions like: 
- "Why do you want the RDKit to support organometallic species?" 
- "Assume we have some way to read/write organometallics, what do you want to do with them?"
- "What kind of species do you want to be able to work with?"

The more concrete you can be the better. If you can provide references or pointers to more information describing your use case(s), please do so. 

A reminder: if we start discussion/arguing about file formats at this stage of the process we will have lost. The time for that discussion will come, but until we know what we want to do with the molecules once we've read them in, there's not much point talking about how we're going to read them in.

**Comment by jhjensen2** [comment]
1. I need to have the correct formal charge of the molecule for QM calculations. It's most important to me that the sum of any atom charges add up to the correct total charge of the molecule. I include the "hydride charge bug" I talked about in this category.
1. I want to do substructure searching
1. I want to manipulate the graph with rxnsmarts, rdmol, etc. to generate new molecules
1. I need ResonanceMolSupplier to work with TMs. I know Paolo made some progress on this in the nightly build but there are still problems last time i checked.
1. I want a canonical representation to store in my database (if ResonanceMolSupplier works correctly, I think I can make something that will work well enough for me, so lower priority).
1. I want to generate fingerprints / Tanimoto similarity.
1. I want to generate reasonable 3D conformers. I only put this last because I found a workaround with Open Babel.

I want to work mostly with organometallic catalysts (typically 1 TM + organic ligands)

**Reply by greglandrum** [reply]
Thanks @jhjensen2!
Would you mind sorting that list so the higher priority items are at the top (in case you haven't done that already)?

**Comment by peonor** [comment]
I would like to be able to create reasonable 3D geometries, dependent on charge and oxidation state. My cases usually contain pi-bound ligands, on square planar metals. A typical case would be .eta.3-allyl on a Pd with two additional neutral ligands, overall the complex would be a cation.

**Comment by throssellqsim** [comment]
I have been using RDKit to generate 3d geometries for organo-metallic systems by decomposing and rebuilding molecules.  If RDKit were to extend its support to organometallics I would like 

1. Point group enforced or constrained 3d geometry generation.
2. Identification and generation of geometric.optical isomers around metal centers from SMILES
3. Support for eta- and dative bonds

If these already exist or there are ways of handling it would I appreciate any information.

**Comment by dkgiesen** [comment]
I work with organometallic complexes with many varied VSEPR geometries for both QM and MD calculations.  The highest priority geometries would be octahedral and square planar, but tetrahedral and trigonal planar geometries are also of interest, coordinations greater than 6 are of significantly less interest.  Ligands can be monodentate, bidentate or haptic.

I would like a way to use SMILES or INCHI to be able to distinguish between conformational isomers of these types of complexes.  For instance, between fac and mer versions of IrPPy3 (https://www.sigmaaldrich.com/catalog/substance/tris2phenylpyridinatoc2niridiumiii654789492886611?lang=en&region=US) or cis vs trans square planar complexes. This would include complexes with haptic ligands and dative bonds.

**Comment by lutzweber** [comment]
Not sure if I fulfill your respectful request above. I would like to recognize dative bonds from images of metal organics and represent them in MOL v3000 or smiles. 

While Cl[Cu](C)(<-O(C)C)(<-O(C)C) works nicely with RDK, for a typical Grignard reagent I[Mg](C)(<-O(C)C)(<-O(C)C) RDKit says:
RDKit ERROR: [18:25:27] Explicit valence for atom # 1 Mg, 4, is greater than permitted
[18:25:27] Explicit valence for atom # 1 Mg, 4, is greater than permitted
>> should I increase the allowed valences for Mg before?

**Comment by greglandrum** [comment]
@lutzweber In theory I would be fine changing the valence states of Mg to 2 (the default) and -1 (allowing the Grignards to be represented), but those kinds of changes can have far-reaching effects and I would want to be sure that it doesn't break anything else.
